src/model/framework/ISearchBase.ts

```diff
@@ -0,0 +1,17 @@
+import type { AutocompleteInteraction } from "discord.js";
+
+export type SearchBase = {
+    name: string;
+    value: string;
+};
+
+export interface ISearchBase<T extends SearchBase> {
+    /**
+     * Preform a search on the search microservice
+     * @param interaction
+     */
+    search(interaction: AutocompleteInteraction): Promise<T[]>;
+}
+
+export const autoCompleteAppUrl = "http://localhost:8083";
```

Comment:
Can we possible set this via `@Property` decorator?

Reply:
yes :)